File: openapi/code_samples/java/api_users_authenticate.java
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/user")
public class UserController {

    @PostMapping("/login")
    public ResponseEntity<Object> login(@RequestBody LoginUser loginUser) {
        // Your authentication logic here
        // Simulated authentication for demonstration purposes
        if (loginUser.getUsername().equals("sampleuser") && loginUser.getPassword().equals("password123")) {
            return ResponseEntity.ok(new UserToken("sampleAuthToken"));
        }

        return ResponseEntity.badRequest().body(new Fault("Invalid username or password."));
    }
}

File: openapi/code_samples/csharp/api_users_authenticate.cs
using Microsoft.AspNetCore.Mvc;
using Swashbuckle.AspNetCore.Annotations;

[ApiController]
[Route("api/[controller]")]
public class UserController : ControllerBase
{
    [HttpPost]
    [Route("login")]
    [SwaggerOperation(Summary = "This is used to authenticate a user.")]
    [ProducesResponseType(typeof(object), 200)]
    [ProducesResponseType(typeof(Fault), 400)]
    public IActionResult Login([FromBody] LoginUser loginUser)
    {
        // Your authentication logic here
        // Simulated authentication for demonstration purposes
        if (loginUser.Username == "sampleuser" && loginUser.Password == "password123")
        {
            var userToken = new { Token = "sampleAuthToken" };
            return Ok(userToken);
        }

        return BadRequest(new Fault { ErrorMessage = "Invalid username or password." });
    }
}

public class LoginUser
{
    public string Username { get; set; }
    public string Password { get; set; }
}
